[JSC] Remove `<?>` from wasm function stack trace when `useEagerWasmModuleHashing` is disabled

sosukesuzuki opened the pull request:
#### ed86232a3c29841b3009f4fd03ada9cfee76cfb1
<pre>
[JSC] Remove `&lt;?&gt;` from wasm function stack trace when `useEagerWasmModuleHashing` is disabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=297880">https://bugs.webkit.org/show_bug.cgi?id=297880</a>

Reviewed by Yusuke Suzuki.

Currently, Wasm function stack traces contain &quot;&lt;?&gt;&quot; as follows:

    run@&lt;?&gt;:wasm-function

This is a placeholder for the module hash that is only displayed when the
useEagerWasmModuleHashing option (disabled by default) is enabled. In other
words, when this option is disabled, it has no meaning.

This patch fixes Wasm stack traces to not include &quot;&lt;?&gt;&quot; when
useEagerWasmModuleHashing is disabled.

* JSTests/stress/sampling-profiler-wasm-name-section.js:
(platformSupportsSamplingProfiler.vm.isWasmSupported):
* JSTests/stress/sampling-profiler-wasm.js:
(platformSupportsSamplingProfiler.vm.isWasmSupported):
* JSTests/wasm/function-tests/nameSection.js:
* JSTests/wasm/function-tests/stack-overflow.js:
(import.Builder.from.string_appeared_here.import.as.assert.from.string_appeared_here.assertOverflows):
(assertOverflows.assertThrows):
(assertOverflows):
* JSTests/wasm/function-tests/stack-trace.js:
* JSTests/wasm/stress/simple-inline-stacktrace-2.js:
(i.catch):
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch-2.js:
(let.readMem.new.Int32Array.wasm_instance.exports.mem.buffer.const.iterCount.9.verifyStack):
(): Deleted.
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch.js:
(let.readMem.new.Int32Array.wasm_instance.exports.mem.buffer.const.iterCount.9.verifyStack):
(): Deleted.
* JSTests/wasm/stress/simple-inline-stacktrace.js:
(i.catch):
* Source/JavaScriptCore/runtime/StackFrame.cpp:
(JSC::StackFrame::sourceURL const):
(JSC::StackFrame::sourceURLStripped const):
* Source/JavaScriptCore/wasm/WasmIndexOrName.h:
(JSC::Wasm::IndexOrName::moduleName const):
* Source/JavaScriptCore/wasm/WasmNameSection.h:
(JSC::Wasm::NameSection::NameSection):

Canonical link: <a href="https://commits.webkit.org/299359@main">https://commits.webkit.org/299359@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da3a2a62ae1c4d17e5fdec72f444a54ae17f7bfd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118020 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37698 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28334 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124170 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70055 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38391 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46280 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89560 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59160 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120972 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30576 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105827 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70052 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29644 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23944 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67837 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110131 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99999 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24122 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127251 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/116530 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44923 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33848 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98233 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45284 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102051 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98019 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25052 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43417 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21406 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41358 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44795 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50469 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145228 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44255 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37360 "Found 1 new JSC binary failure: testapi, Found 20034 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/concat1.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47600 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45944 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
Is it aligned to the other engines and the spec's conventional format? If so, r=me

sosukesuzuki committed:
[JSC] Remove `<?>` from wasm function stack trace when `useEagerWasmModuleHashing` is disabled
https://bugs.webkit.org/show_bug.cgi?id=297880

Reviewed by Yusuke Suzuki.

Currently, Wasm function stack traces contain "<?>" as follows:

    run@<?>:wasm-function

This is a placeholder for the module hash that is only displayed when the
useEagerWasmModuleHashing option (disabled by default) is enabled. In other
words, when this option is disabled, it has no meaning.

This patch fixes Wasm stack traces to not include "<?>" when
useEagerWasmModuleHashing is disabled.

* JSTests/stress/sampling-profiler-wasm-name-section.js:
(platformSupportsSamplingProfiler.vm.isWasmSupported):
* JSTests/stress/sampling-profiler-wasm.js:
(platformSupportsSamplingProfiler.vm.isWasmSupported):
* JSTests/wasm/function-tests/nameSection.js:
* JSTests/wasm/function-tests/stack-overflow.js:
(import.Builder.from.string_appeared_here.import.as.assert.from.string_appeared_here.assertOverflows):
(assertOverflows.assertThrows):
(assertOverflows):
* JSTests/wasm/function-tests/stack-trace.js:
* JSTests/wasm/stress/simple-inline-stacktrace-2.js:
(i.catch):
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch-2.js:
(let.readMem.new.Int32Array.wasm_instance.exports.mem.buffer.const.iterCount.9.verifyStack):
(): Deleted.
* JSTests/wasm/stress/simple-inline-stacktrace-with-catch.js:
(let.readMem.new.Int32Array.wasm_instance.exports.mem.buffer.const.iterCount.9.verifyStack):
(): Deleted.
* JSTests/wasm/stress/simple-inline-stacktrace.js:
(i.catch):
* Source/JavaScriptCore/runtime/StackFrame.cpp:
(JSC::StackFrame::sourceURL const):
(JSC::StackFrame::sourceURLStripped const):
* Source/JavaScriptCore/wasm/WasmIndexOrName.h:
(JSC::Wasm::IndexOrName::moduleName const):
* Source/JavaScriptCore/wasm/WasmNameSection.h:
(JSC::Wasm::NameSection::NameSection):

Canonical link: https://commits.webkit.org/299359@main
